diff --git a/target/linux/mpc85xx/Makefile b/target/linux/mpc85xx/Makefile
index a32be90..81c7afd 100644
--- a/target/linux/mpc85xx/Makefile
+++ b/target/linux/mpc85xx/Makefile
@@ -9,7 +9,7 @@ include $(TOPDIR)/rules.mk
 ARCH:=powerpc
 BOARD:=mpc85xx
 BOARDNAME:=Freescale MPC85xx
-CPU_TYPE:=8540
+CPU_TYPE:=8548
 FEATURES:=squashfs
 MAINTAINER:=Imre Kaloz <kaloz@openwrt.org>
 SUBTARGETS=generic p1020
@@ -26,7 +26,7 @@ endif
 DEFAULT_PACKAGES += \
 	kmod-input-core kmod-input-gpio-keys kmod-button-hotplug \
 	kmod-leds-gpio kmod-booke-wdt \
-	swconfig kmod-ath9k wpad-mini
+	swconfig kmod-ath9k wpad
 
 define Target/Description
 	Build images for the Freescale MPC85xx based boards.
