diff --git a/target/linux/mpc85xx/Makefile b/target/linux/mpc85xx/Makefile
index 161c8cb..94db5e3 100644
--- a/target/linux/mpc85xx/Makefile
+++ b/target/linux/mpc85xx/Makefile
@@ -9,7 +9,7 @@ include $(TOPDIR)/rules.mk
 ARCH:=powerpc
 BOARD:=mpc85xx
 BOARDNAME:=Freescale MPC85xx
-CPU_TYPE:=8540
+CPU_TYPE:=8540
 FEATURES:=squashfs
 MAINTAINER:=Imre Kaloz <kaloz@openwrt.org>
 SUBTARGETS:=generic p1020
@@ -29,6 +29,6 @@ endif
 
 DEFAULT_PACKAGES += \
 	kmod-input-core kmod-input-gpio-keys kmod-button-hotplug \
-	kmod-leds-gpio swconfig kmod-ath9k wpad-mini
+	kmod-leds-gpio swconfig kmod-ath9k wpad
 
 $(eval $(call BuildTarget))
diff --git a/target/linux/mpc85xx/profiles/tp-link.mk b/target/linux/mpc85xx/profiles/tp-link.mk
index 18a7cdd..3f8fbb8 100644
--- a/target/linux/mpc85xx/profiles/tp-link.mk
+++ b/target/linux/mpc85xx/profiles/tp-link.mk
@@ -7,7 +7,7 @@
 
 define Profile/TLWDR4900
 	NAME:=TP-Link TL-WDR4900
-	PACKAGES:=kmod-ath9k wpad-mini
+	PACKAGES:=kmod-ath9k wpad
 endef
 
 define Profile/TLWDR4900/Description
