--- a/package/qca/qca-ssdk/Makefile	2017-03-24 13:12:16.366306220 +0800
+++ b/package/qca/qca-ssdk/Makefile	2017-03-24 13:02:11.254733089 +0800
@@ -64,7 +64,7 @@ QCASSDK_CONFIG_OPTS+= TOOL_PATH=$(TOOLCH
 
 ifeq ($(LOCAL_VARIANT),hnat)
     QCASSDK_CONFIG_OPTS+= HNAT_FEATURE=enable
-ifeq ($(CONFIG_TARGET_ar71xx),)
+ifeq ($(CONFIG_TARGET_mpc85xx),)
     QCASSDK_CONFIG_OPTS+= RFS_FEATURE=enable
 endif
 endif
--- a/package/qca/qca-ssdk/src/make/config.mk	2017-03-23 23:32:15.000000000 +0800
+++ b/package/qca/qca-ssdk/src/make/config.mk	2017-03-24 13:21:08.603904543 +0800
@@ -11,7 +11,7 @@ endif
 
 #define cpu type such as PPC MIPS ARM X86
 ifndef CPU
-  CPU=mips
+  CPU=ppc
 endif
 
 #define os type such as linux netbsd vxworks
@@ -59,7 +59,7 @@ else
      ifneq (TRUE, $(FAL))
          $(error FAL must be TRUE when CHIP_TYPE is defined as ALL_CHIP!)
      endif
-     SUPPORT_CHIP = ISIS ISISC SHIVA DESS
+     SUPPORT_CHIP = ISIS ISISC SHIVA
   endif
 
   ifndef SUPPORT_CHIP
--- a/package/qca/qca-ssdk/src/make/linux_opt.mk	2017-03-23 23:32:15.000000000 +0800
+++ b/package/qca/qca-ssdk/src/make/linux_opt.mk	2017-03-24 12:41:53.237165048 +0800
@@ -267,29 +267,22 @@ ifeq (KSLIB, $(MODULE_TYPE))
 	          -include $(SYS_PATH)/include/linux/kconfig.h
             else
             MODULE_INC += -I$(SYS_PATH) \
-	      -I$(TOOL_PATH)/../lib/gcc/mips-openwrt-linux-$(TARGET_SUFFIX)/$(GCC_VERSION)/include/ \
+	      -I$(TOOL_PATH)/../lib/gcc/mpc85xx-openwrt-linux-$(TARGET_SUFFIX)/$(GCC_VERSION)/include/ \
               -I$(SYS_PATH)/include \
 	      -I$(SYS_PATH)/source/include \
-              -I$(SYS_PATH)/source/arch/mips/mach-msm/include \
-	      -I$(SYS_PATH)/arch/mips/mach-msm/include \
-	      -I$(SYS_PATH)/source/arch/mips/include \
-              -I$(SYS_PATH)/arch/mips/include \
-	      -I$(SYS_PATH)/source/arch/mips/include/asm \
-	      -I$(SYS_PATH)/arch/mips/include/generated \
+	      -I$(SYS_PATH)/source/arch/powerpc/include \
+              -I$(SYS_PATH)/arch/powerpc/include \
+	      -I$(SYS_PATH)/source/arch/powerpc/include/asm \
+	      -I$(SYS_PATH)/arch/powerpc/include/generated \
               -I$(SYS_PATH)/include/generated/uapi \
 	      -I$(SYS_PATH)/include/uapi \
-	      -I$(SYS_PATH)/arch/mips/include/uapi \
-	      -I$(SYS_PATH)/source/arch/mips/include/asm/mach \
 	      -include $(SYS_PATH)/include/linux/kconfig.h \
-              -I$(SYS_PATH)/arch/mips/include/asm/mach-ar7240 \
-              -I$(SYS_PATH)/arch/mips/include/asm/mach-generic \
-              -I$(SYS_PATH)/arch/mips/include/asm/mach-ar7 \
               -I$(SYS_PATH)/usr/include
 
 	    #CPU_CFLAG    = -G 0 -mno-abicalls -fno-pic -pipe -mabi=32 -march=mips32r2
             ifndef CPU_CFLAG
-            CPU_CFLAG    = -Wstrict-prototypes -fomit-frame-pointer -G 0 -mno-abicalls -fno-strict-aliasing \
-			-O2 -fno-pic -pipe -mabi=32 -march=mips32r2 -DMODULE -mlong-calls -DEXPORT_SYMTAB
+            CPU_CFLAG    = -Wstrict-prototypes -fomit-frame-pointer -G 0 -fno-strict-aliasing -fno-common \
+			-O2 -fno-pic -pipe -mcpu=8548 -DMODULE -mlongcall -DEXPORT_SYMTAB
             endif
             endif
 
